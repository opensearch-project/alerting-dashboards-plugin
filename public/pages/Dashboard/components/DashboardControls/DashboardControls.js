/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { EuiFieldSearch, EuiFlexGroup, EuiSelect, EuiFlexItem, EuiPagination } from '@elastic/eui';
import { ALERT_STATE } from '../../../../utils/constants';

const severityOptions = [
  { value: 'ALL', text: 'All severity levels' },
  { value: '1', text: '1 (Highest)' },
  { value: '2', text: '2 (High)' },
  { value: '3', text: '3 (Medium)' },
  { value: '4', text: '4 (Low)' },
  { value: '5', text: '5 (Lowest)' },
];

const stateOptions = [
  { value: 'ALL', text: 'All alerts' },
  { value: ALERT_STATE.ACTIVE, text: 'Active' },
  { value: ALERT_STATE.ACKNOWLEDGED, text: 'Acknowledged' },
  { value: ALERT_STATE.COMPLETED, text: 'Completed' },
  { value: ALERT_STATE.ERROR, text: 'Error' },
  { value: ALERT_STATE.DELETED, text: 'Deleted' },
];

const DashboardControls = ({
  activePage,
  pageCount,
  search,
  severity = severityOptions[0],
  state = stateOptions[0],
  onSearchChange,
  onSeverityChange,
  onStateChange,
  onPageChange,
  isAlertsFlyout = false,
}) => (
  <EuiFlexGroup style={{ padding: '0px 5px' }}>
    <EuiFlexItem>
      <EuiFieldSearch
        fullWidth={true}
        placeholder="Search"
        onChange={onSearchChange}
        value={search}
      />
    </EuiFlexItem>

    {isAlertsFlyout ? null : (
      <EuiFlexItem grow={false}>
        <EuiSelect options={severityOptions} value={severity} onChange={onSeverityChange} />
      </EuiFlexItem>
    )}

    <EuiFlexItem grow={false}>
      <EuiSelect options={stateOptions} value={state} onChange={onStateChange} />
    </EuiFlexItem>
    <EuiFlexItem grow={false} style={{ justifyContent: 'center' }}>
      <EuiPagination pageCount={pageCount} activePage={activePage} onPageClick={onPageChange} />
    </EuiFlexItem>
  </EuiFlexGroup>
);

export default DashboardControls;
