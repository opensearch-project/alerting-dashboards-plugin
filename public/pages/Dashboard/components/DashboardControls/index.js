/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import DashboardControls from './DashboardControls';

export default DashboardControls;
