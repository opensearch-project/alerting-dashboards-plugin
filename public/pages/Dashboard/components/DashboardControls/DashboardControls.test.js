/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from 'enzyme';

import DashboardControls from './DashboardControls';

describe('DashboardControls', () => {
  test('renders', () => {
    const component = (
      <DashboardControls
        activePage={1}
        pageCount={2}
        search={''}
        severity={'ALL'}
        state={'ALL'}
        onSearchChange={() => {}}
        onSeverityChange={() => {}}
        onStateChange={() => {}}
        onPageChange={() => {}}
      />
    );
    expect(render(component)).toMatchSnapshot();
  });
});
