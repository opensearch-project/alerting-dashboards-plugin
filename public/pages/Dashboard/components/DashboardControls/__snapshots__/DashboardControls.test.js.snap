// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardControls renders 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--directionRow euiFlexGroup--responsive"
  style="padding:0px 0px 16px"
>
  <div
    class="euiFlexItem"
  >
    <div
      class="euiFormControlLayout euiFormControlLayout--fullWidth euiFormControlLayout--compressed"
    >
      <div
        class="euiFormControlLayout__childrenWrapper"
      >
        <input
          class="euiFieldSearch euiFieldSearch--fullWidth euiFieldSearch--compressed"
          placeholder="Search"
          type="search"
          value=""
        />
        <div
          class="euiFormControlLayoutIcons"
        >
          <span
            class="euiFormControlLayoutCustomIcon"
          >
            <div>
              EuiIconMock
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
  >
    <div
      class="euiFormControlLayout euiFormControlLayout--compressed"
    >
      <div
        class="euiFormControlLayout__childrenWrapper"
      >
        <select
          class="euiSelect euiSelect--compressed"
        >
          <option
            selected=""
            value="ALL"
          >
            All severity levels
          </option>
          <option
            value="1"
          >
            1 (Highest)
          </option>
          <option
            value="2"
          >
            2 (High)
          </option>
          <option
            value="3"
          >
            3 (Medium)
          </option>
          <option
            value="4"
          >
            4 (Low)
          </option>
          <option
            value="5"
          >
            5 (Lowest)
          </option>
        </select>
        <div
          class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
        >
          <span
            class="euiFormControlLayoutCustomIcon"
          >
            <div>
              EuiIconMock
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
  >
    <div
      class="euiFormControlLayout euiFormControlLayout--compressed"
    >
      <div
        class="euiFormControlLayout__childrenWrapper"
      >
        <select
          class="euiSelect euiSelect--compressed"
          data-test-subj="dashboardAlertStateFilter"
        >
          <option
            selected=""
            value="ALL"
          >
            All alerts
          </option>
          <option
            value="ACTIVE"
          >
            Active
          </option>
          <option
            value="ACKNOWLEDGED"
          >
            Acknowledged
          </option>
          <option
            value="COMPLETED"
          >
            Completed
          </option>
          <option
            value="ERROR"
          >
            Error
          </option>
          <option
            value="DELETED"
          >
            Deleted
          </option>
        </select>
        <div
          class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
        >
          <span
            class="euiFormControlLayoutCustomIcon"
          >
            <div>
              EuiIconMock
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
